reporter/attrmgr.go

```diff
@@ -0,0 +1,62 @@
+package reporter
+
+import (
+	"fmt"
+
+	common "go.opentelemetry.io/proto/otlp/common/v1"
+	profiles "go.opentelemetry.io/proto/otlp/profiles/v1experimental"
+)
+
+// AttrIndex is an index in the `Profile.attribute_table`.
+type AttrIndex = uint64
+
+// AttrTableManager maintains the `Profile.attribute_table` field for the
+// `profile.Profile` that it was created for. Attributes are automatically
+// deduplicated by the manager.
+type AttrTableManager struct {
+	// indices maps compound keys to the indices in the attribute table.
+	indices map[string]AttrIndex
+
+	// profile being built right now.
+	profile *profiles.Profile
+}
+
+func NewAttrTableManager(profile *profiles.Profile) *AttrTableManager {
+	return &AttrTableManager{
+		indices: make(map[string]AttrIndex),
+		profile: profile,
+	}
+}
+
+func (m *AttrTableManager) AddIntAttr(key string, value int64) AttrIndex {
+	compound := fmt.Sprintf("%v_%d", key, value)
+	val := common.AnyValue{Value: &common.AnyValue_IntValue{IntValue: value}}
+	return m.addAnyAttr(key, compound, &val)
+}
+
+func (m *AttrTableManager) AddStringAttr(key, value string) AttrIndex {
```

Comment:
How would skipping empty strings work here (as suggested in #233)? We can not just return a 0 index.

Reply:
I was under the impression that the discussion in #233 was leaning towards not doing the filtering in the function/component responsible for managing the attribute table. In the interest of not spreading the discussion over two issues, I'll let this PR wait until #233 is either merged. If we decide on letting the manager fn filter, we can simply rework this to:

```go
AppendStringSttr(attrs []AttrIndex, key, value) []AttrIndex
```

Reply:
Implemented this now, but with pointer arg (`attrs *[]AttrIndex`) that is mutated in place instead of returning the extended slice.